storage/storage_transport.go

```diff
@@ -0,0 +1,294 @@
+			// recognize.
+			return nil, ErrInvalidReference
+		}
+	}
+	return s.ParseStoreReference(store, reference)
+}
+
+func (s storageTransport) GetStoreImage(store storage.Store, ref types.ImageReference) (*storage.Image, error) {
+	dref := ref.DockerReference()
+	if dref == nil {
+		if sref, ok := ref.(*storageReference); ok {
+			if sref.id != "" {
+				if img, err := store.GetImage(sref.id); err == nil {
+					return img, nil
+				}
+			}
+		}
+		return nil, ErrInvalidReference
+	}
+	return store.GetImage(verboseName(dref))
+}
+
+func (s *storageTransport) GetImage(ref types.ImageReference) (*storage.Image, error) {
+	store, err := s.GetStore()
+	if err != nil {
+		return nil, err
+	}
+	return s.GetStoreImage(store, ref)
+}
+
+func (s storageTransport) ValidatePolicyConfigurationScope(scope string) error {
+	// Check if there's a store location prefix.
+	if scope[0] == '[' {
+		close := strings.IndexRune(scope, ']')
+		if close < 1 {
+			return ErrInvalidReference
+		}
+		storeSpec := scope[1:close]
+		scope = scope[close+1:]
+		storeInfo := strings.SplitN(storeSpec, "@", 2)
+		if len(storeInfo) == 1 && storeInfo[0] != "" {
+			// One component: the graph root.
+			if !filepath.IsAbs(storeInfo[0]) {
+				return ErrPathNotAbsolute
+			}
+		} else if len(storeInfo) == 2 && storeInfo[0] != "" && storeInfo[1] != "" {
+			// Two components: the driver type and the graph root.
+			if !filepath.IsAbs(storeInfo[1]) {
+				return ErrPathNotAbsolute
+			}
+		} else {
+			// Anything else: store specified in a form we don't
+			// recognize.
+			return ErrInvalidReference
+		}
+	}
+	// That might be all of it, and that's okay.
+	if scope == "" {
+		return nil
+	}
+	// But if there is anything left, it has to be a name, with or without
+	// an ID.
+	if scope[0] == '@' {
+		return ErrInvalidReference
+	}
+	scopeInfo := strings.SplitN(scope, "@", 2)
+	if len(scopeInfo) == 1 && scopeInfo[0] != "" {
+		id, name, err := reference.ParseIDOrReference(scopeInfo[0])
+		if err != nil {
+			return err
+		}
+		if id.Validate() == nil {
+			return ErrInvalidReference
+		}
+		if name == nil {
+			return ErrInvalidReference
+		}
+	} else if len(scopeInfo) == 2 && scopeInfo[0] != "" && scopeInfo[1] != "" {
+		id, name, err := reference.ParseIDOrReference(scopeInfo[0])
+		if err != nil {
+			return err
+		}
+		if id.Validate() == nil {
+			return ErrInvalidReference
+		}
+		if name == nil {
+			return ErrInvalidReference
+		}
+		id, name, err = reference.ParseIDOrReference(scopeInfo[1])
+		if err != nil {
+			return err
+		}
+		if id.Validate() != nil {
+			return ErrInvalidReference
+		}
+		if name != nil {
+			return ErrInvalidReference
+		}
+	} else {
+		return ErrInvalidReference
+	}
+	return nil
+}
+
+func verboseName(name reference.Named) string {
+	name = reference.WithDefaultTag(name)
+	tag := ""
+	if tagged, ok := name.(reference.NamedTagged); ok {
+		tag = tagged.Tag()
+	}
+	return name.FullName() + ":" + tag
```

Comment:
Hmm, changing the return type for `ParseStoreReference()` to return a `storageReference` should do the trick there, since `GetStoreImage()` is mainly there to allow us to find a stored image using the reference that we passed to `image.Copy()`.  Changing it.